src/webpack/loaderUtils/configHelpers.js

```diff
@@ -34,38 +29,42 @@ const getNamespace = (directory, namespace) => {
 /**
  * Returns an array of routes based on folder path.
  */
-const getRoutes = (directory, type, fileName, entryPoint) => {
-  // Remove the directories up to the entry point.
-  const modifiedDirectory = directory.replace(entryPoint, '');
+const getRoutes = (fileName, entryPoint) => {
+  const parsedPath = path.parse(entryPoint);
 
-  let routes = modifiedDirectory.split('/');
+  let routes = [];
+  if (parsedPath.dir !== '/') {
+    routes = parsedPath.dir.split('/');
+  }
 
-  // Note: spliting on seperator results in the first array element to be '' so we shift to get rid of it.
+  // Note: splitting on separator results in the first array element to be '' so we shift to get rid of it.
   routes.shift();
 
-  // Trim the first folder after entrypoints if it is named the same as the page type.
-  if ((routes[0] || '').toUpperCase() === type.toUpperCase()) {
-    routes = routes.slice(1);
-  }
-
   // add on the file name as the last route
   routes.push(fileName);
 
   return routes;
 };
 
-/** Returns an object of the end most extention and the filename minus that extension.
- * This may be used mulitiple times on a string to retirve all extensions.
+/** Returns an object of the end most extension and the filename minus that extension.
+ * This may be used multiple times on a string to retrieve all extensions.
  */
-const parseExtension = fileName => ({
-  name: fileName.replace(/\.[^.]+$/, ''),
-  extension: /[^.]+$/.exec(fileName)[0],
-});
+const parseExtension = fileName => {
+  const result = /\.([^.]+$)/.exec(fileName);
+
+  let extension;
+  if (result) {
+    [, extension] = result;
```

Comment:
before this was pulling the first value? not the second? is that right?

Reply:
it was a bug previously, we want the first capture group here not the full regex match.

Reply:
Hey now, @StephenEsser has been big on doc everything lately and it does make it easier once you've been away from it for a while.